main.js

```diff
@@ -118,33 +62,36 @@ const editListItem = (event, id) => {
     })
 }
 
-
 const filterList = event => {
     const currentNodeList = document.querySelectorAll('.list-item')
-    const finishedTasksList = todos.filter(todo => todo.isDone)
-    const finishedId = finishedTasksList.map(task => task.id)
-    console.log(finishedId)
-    console.log(finishedTasksList)
-    console.log(currentNodeList)
+    const finishedTasksIds = todos
+        .filter(todo => todo.getIsDone())
+        .map(task => task.getId())
 
     if (event.target.id === 'todo') {
         currentNodeList.forEach(div => {
-            if (finishedId.includes(div.getAttribute('todo-id'))) {
+            if (finishedTasksIds
```

Comment:
```
if (event.target.id === 'todo') 
```
looks bad

Reply:
I have no idea how to replace this. The function has not relation with a Todo class

Reply:
Simply add 3 event listeners instead of one:
```
onShowAllTodos
onShowDoneTodos
onShowNotDoneTodos
```